code/handler.ts

```diff
@@ -0,0 +1,130 @@
+    },
+    "/v0/files/{id}": {
+        GET: getFile,
+        POST: createFile,
+        DELETE: deleteFile
+    }
+}
+
+export async function handler(
+    event: APIGatewayProxyEvent,
+    context: Context
+): Promise<APIGatewayProxyResult> {
+    const methodHandlerMap: MethodHandlerMap = resourceHandlerMap[event.resource] || {}
+    const methodHandler: ApiGatewayProxyHandler = methodHandlerMap[event.httpMethod] || handleRouteNotFound
+
+    try {
+        return await methodHandler(event, context)
+    } catch(err) {
+        if(err instanceof ValidationError) {
+            return jsonResponse(HttpStatus.BadRequest, { error: err.message })
+        } else {
+            console.error(err)
+            return jsonResponse(HttpStatus.ServerError, { error: 'Unexpected server error' })
+        }
+    }
+}
+
+function getS3BucketName(): string {
+    if (process.env.S3_BUCKET) return process.env.S3_BUCKET
+
+    throw new Error("S3_BUCKET environment variable has not been specified")
+}
+
+function extractFileName(event: APIGatewayProxyEvent): string {
+    const fileName = event.pathParameters?.id
+    if(!fileName) throw new ValidationError('file name must be specified')
+    return fileName
+}
+
+function jsonResponse(statusCode: HttpStatus, body: any): APIGatewayProxyResult {
+    return {
+        statusCode,
+        headers: { 'content-type': 'application/json' },
+        body: JSON.stringify(body)
+    }
+}
+
+function mapAwsErrorToResponse(err: any) {
+    if (err.code && err.code === 'NoSuchKey') {
+        return jsonResponse(HttpStatus.NotFound, { error: 'no such file exists' })
+    }
+    throw err
+}
+
+async function handleRouteNotFound(
+    event: APIGatewayProxyEvent,
+    context: Context
+): Promise<APIGatewayProxyResult> {
+    return jsonResponse(HttpStatus.NotFound, { error: 'route not found' })
+}
+
+async function listFiles(
+    event: APIGatewayProxyEvent,
+    context: Context
+): Promise<APIGatewayProxyResult> {
+    const result = await s3.listObjects({
+        Bucket: getS3BucketName()
+    }).promise()
+
+    return jsonResponse(HttpStatus.OK, { files: result.Contents?.map(content => content.Key) })
+}
+
+async function getFile(
+    event: APIGatewayProxyEvent,
+    context: Context
+): Promise<APIGatewayProxyResult> {
+    try {
+        const result = await s3.getObject({
+            Bucket: getS3BucketName(),
+            Key: extractFileName(event)
+        }).promise()
+
+        return {
+            statusCode: HttpStatus.OK,
+            headers: { 'content-type': '' },
+            body: result.Body?.toString() || ''
+        }
+    } catch(err: any) {
+        return mapAwsErrorToResponse(err)
+    }
+}
+
+async function createFile(
+    event: APIGatewayProxyEvent,
+    context: Context
+): Promise<APIGatewayProxyResult> {
+    await s3.putObject({
+        Bucket: getS3BucketName(),
+        Key: extractFileName(event),
+        Body: `This file was created on ${new Date().toISOString()}`
```

Comment:
I thought we actually needed create a file from the payload being sent in the body.
In case we were supposed to do so, I believe this could be achieved by using a content type such as `multipart/form-data`. But I don't think is necessary, since you have already proved that you know how to use CDK & Typescript, which was the purpose of the challenge.

Reply:
Yeah, I thought about that, but the main focus for me was just to generate the AWS resources